### PR TITLE
feat(ci): New Docker AGW AMI And GHZ AGW load test

### PR DESCRIPTION
**Pull request by quentinDERORY**
Signed-off-by: quentinDERORY <15911421+quentinDERORY@users.noreply.github.com>

## Summary

Generate a new AWS AMI for launching AGW with docker
GHA job to launch AGW docker load test in AWS

## Test Plan

Test on PR 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: .github/workflows/agw-docker-load-test.yaml
Comment (arunuke) [inline .github/workflows/agw-docker-load-test.yaml]: Is there a way to trigger newer builds of the base images based on files that have changed? Alternatively, we do nightly builds and rotate every 7 days or so? Should not hold this PR back, but would be good to capture this somewhere.

File: .github/workflows/ami-docker-workflow.yml
Comment (arunuke) [inline .github/workflows/ami-docker-workflow.yml]: Would this add a date/commit id based name tag to the image? This would help in keeping an array of images that we can rotate based on an arbitrary number. I see that the value is currently set to "latest"?

File: experimental/cloudstrapper/playbooks/roles/docker-platform/tasks/magma-repo.yaml
Comment (arunuke) [inline experimental/cloudstrapper/playbooks/roles/docker-platform/tasks/magma-repo.yaml:11]: The agw_install_docker script appears to remove anything in the /opt/magma directory and clone it. I agree that doing it here makes sense too, but we might have to refactor it.

**Review by arunuke — CHANGES_REQUESTED** [review]
A general comment is to address some of the yamlint warnings (end of file, true/false etc.). LGTM otherwise if the labeling changes can be considered.

File: .github/workflows/agw-docker-load-test.yaml
Comment (quentinDERORY) [reply .github/workflows/agw-docker-load-test.yaml]: yes I can make the snapshot docker ami workflow use a trigger based on file changes, what files do you have in mind?

File: .github/workflows/ami-docker-workflow.yml
Comment (quentinDERORY) [reply .github/workflows/ami-docker-workflow.yml]: yes still need to parametrize the ami to have the sha!

File: experimental/cloudstrapper/playbooks/roles/docker-platform/tasks/magma-repo.yaml
Comment (quentinDERORY) [reply experimental/cloudstrapper/playbooks/roles/docker-platform/tasks/magma-repo.yaml:11]: Yes best would be to just checkout that script! I will add the install of docker in this script in this PR too.

Comment (quentinDERORY) [reply experimental/cloudstrapper/playbooks/roles/docker-platform/tasks/magma-repo.yaml:11]: might be best to transform that docker intstall script in an ansible role later to avoid checking out the repo here

**Review by ardzoht — APPROVED** [review]
lgtm

File: .github/workflows/agw-docker-load-test.yaml
Comment (arunuke) [inline .github/workflows/agw-docker-load-test.yaml:24]: We would need a plan to make this customizable. Maybe look for a name tag?

**Review by arunuke — APPROVED** [review]
Can you please file an issue for custom AMI id (either passed as a parameter, or picked up using a tag) for the review comment? LGTM otherwise.

File: .github/workflows/agw-docker-load-test.yaml
Comment (quentinDERORY) [reply .github/workflows/agw-docker-load-test.yaml:24]: yes we have similar issue with dev container and bazel, I would like to find something that could fit both use cases.

Comment (quentinDERORY) [reply .github/workflows/agw-docker-load-test.yaml:24]: Tracking it here https://github.com/magma/magma/issues/11113

**Pull request merged**
